test-two/src/services/github.js

```diff
@@ -1,16 +1,64 @@
-import { get } from "../helpers/http";
+import { get, joinPaginatedRequests, post } from "../helpers/http";
 
-const getAuthenticationQueryString = async apiAuthenticationToken =>
-  `?access_token=${apiAuthenticationToken}`;
+const PAGE_SIZE = 100;
+
+const getAuthHeader = apiToken => ({
+  Authorization: `Bearer ${apiToken}`,
+});
 
 export const getUsersForOrganisation = async (
   baseUrl,
+  apiToken,
+  orgId
+) => {
+  const totalUsers =
+    await getTotalUsers(apiToken, orgId);
+
+  return await joinPaginatedRequests(
+    totalUsers,
+    PAGE_SIZE,
+    (pageSize, current) => {
+      return get(
+      `${baseUrl}orgs/${orgId}/members?per_page=${pageSize}&page=${current}`, {
+        headers: {
+          'Content-Type': 'application/json',
+          ...getAuthHeader(apiToken),
+        },
+      }
+    )}
+  )
+};
+
+const getTotalUsers = async (
   apiAuthenticationToken,
   organisationId
 ) => {
-  return get(
-    `${baseUrl}orgs/${organisationId}/members${await getAuthenticationQueryString(
-      apiAuthenticationToken
-    )}`
-  );
-};
+  
+    const response = await post('https://api.github.com/graphql', {
+      headers: {
+        'Content-Type': 'application/json',
+        ...getAuthHeader(apiAuthenticationToken),
+      },
+      body: JSON.stringify({
+        query: `query {
+          organization(login: "${organisationId}") {
+            membersWithRole {
+              totalCount
+            }
+          }
+        }
+      `})
+    });  
+    
+    if(
+      response
+      && response.data
+      && response.data.organization
+      && response.data.organization.membersWithRole
+      && response.data.organization.membersWithRole.totalCount
```

Comment:
Might we consider using a helper to consolidate this into one check? eg the get() method from object-path https://www.npmjs.com/package/object-path

Reply:
I didnt want to change your transpiler in that branch. I wanted to show you that I can work with the environment you provide. I the other branch, the one with TS, it has Optional chaining (?.), and looks a bit better